pkg/platform/common/vmlayer/vmprovider.go

```diff
@@ -560,6 +562,120 @@ func (v *VMPlatform) initRootLB(ctx context.Context, platformConfig *platform.Pl
 	return nil
 }
 
+// updateAppInstConfigForLb
+func (v *VMPlatform) updateAppInstConfigForLb(ctx context.Context, caches *platform.Caches, appInst *edgeproto.AppInst) {
+	log.SpanLog(ctx, log.DebugLevelInfra, "Update rootLb for appInst", "AppInst", appInst.Key)
+	app := edgeproto.App{}
+	if !caches.AppCache.Get(&appInst.AppKey, &app) {
+		log.SpanLog(ctx, log.DebugLevelInfra, "upgrade version single cluster config dir, App not found", "AppInst", appInst.Key)
+		return
+	}
+	if app.Deployment != cloudcommon.DeploymentTypeKubernetes {
+		log.SpanLog(ctx, log.DebugLevelInfra, "not a k8s appinst")
+		return
+	}
+	cinst := edgeproto.ClusterInst{}
+	if !caches.ClusterInstCache.Get(appInst.ClusterInstKey(), &cinst) {
+		log.SpanLog(ctx, log.DebugLevelInfra, "clusterInstNot found", "AppInst", appInst.Key)
+		return
+	}
+	// Only update shared access appInsts
+	if cinst.IpAccess != edgeproto.IpAccess_IP_ACCESS_SHARED {
+		log.SpanLog(ctx, log.DebugLevelInfra, "appinst uses dedicated ip access", "AppInst", appInst.Key)
+		return
+	}
+	appInstFlavor := edgeproto.Flavor{}
+	if !caches.FlavorCache.Get(&appInst.Flavor, &appInstFlavor) {
+		log.SpanLog(ctx, log.DebugLevelInfra, "flavor not found", "AppInst", appInst.Key)
+		return
+	}
+
+	names, err := k8smgmt.GetKubeNames(&cinst, &app, appInst)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "update appinst data for lb, names failed", "AppInst", appInst.Key, "err", err)
+		return
+	}
+	client, err := v.GetClusterPlatformClient(ctx, &cinst, cloudcommon.ClientTypeRootLB)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "update appinst data for lb, get client failed", "AppInst", appInst.Key, "err", err)
+		return
+	}
+
+	// Update appinst manifests on the rootLb
+	mf, err := cloudcommon.GetDeploymentManifest(ctx, v.VMProperties.CommonPf.PlatformConfig.AccessApi, app.DeploymentManifest)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "failed to get the deployment manifest", "AppInst", appInst.Key, "err", err)
+		return
+	}
+	mf, err = k8smgmt.MergeEnvVars(ctx, v.VMProperties.CommonPf.PlatformConfig.AccessApi, &app, appInst, mf, names.ImagePullSecrets, names, &appInstFlavor)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "failed to merge env vars", "AppInst", appInst.Key, "err", err)
+		return
+	}
+	configDir := k8smgmt.GetConfigDirName(names)
+	configName := k8smgmt.GetConfigFileName(names, appInst)
+	err = pc.CreateDir(ctx, client, configDir, pc.NoOverwrite, pc.NoSudo)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "unable to create a directory", "AppInst", appInst.Key, "err", err)
+		return
+	}
+	file := configDir + "/" + configName
+	log.SpanLog(ctx, log.DebugLevelInfra, "writing config file", "file", file, "kubeManifest", mf)
+	err = pc.WriteFile(client, file, mf, "K8s Deployment", pc.NoSudo)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "unable to write manifest", "AppInst", appInst.Key, "err", err)
```

Comment:
This same set of code is duplicated in k8smgmt.createOrUpdateAppInst() (although it's actually missing the network policy part). I would suggest you break out the existing code in k8smgmt.createOrUpdateAppInst() into a sub function like EnsureDeploymentManifest(), so that it can be called from both there and here, to avoid duplicating code. The new function should be idempotent. Also that way you don't need to export GetConfigDirName()/GetConfigFileName(), as it remains an implementation detail of the k8smgmt package.

Reply:
With respect to network policy - I didn't add it, since we don't create multi-tenant clusters, but probably a good idea to add it, since we support them.

Reply:
actually, this is something that should only be done if creating a new cluster, but in this case, we are re-building state, so network policy would already be added to the cluster.